### PR TITLE
[1.26] Deprecate BACKOFF_MAX to introduce DEFAULT_BACKOFF_MAX

### PR DESCRIPTION
**Pull request by pyrooka**
This PR is the deprecation part of the following one: https://github.com/urllib3/urllib3/pull/2494

### CODE REVIEW

**Comment by pyrooka** [comment]
I see no issues when I check the codecov report, but it's failing. Any idea what to do?

**Comment by sethmlarson** [comment]
@pyrooka Don't worry about Codecov, it's really unreliable on the commit status. We can make sure things are okay when we review them.

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Sorry, I started this review but didn't hit submit. Thanks for this!

File: src/urllib3/util/retry.py [hunk 1]
```diff
@@ -69,6 +69,24 @@ def DEFAULT_REDIRECT_HEADERS_BLACKLIST(cls, value):
...
+            "Using 'Retry.BACKOFF_MAX' is deprecated and "
+            "will be removed in v2.0. Use 'Retry.DEFAULT_BACKOFF_MAX' instead",
+            DeprecationWarning,
+        )
+        return cls.DEFAULT_BACKOFF_MAX
+
+    @BACKOFF_MAX.setter
+    def BACKOFF_MAX(cls, value):
+        warnings.warn(
+            "Using 'Retry.DEFAULT_METHOD_WHITELIST' is deprecated and "
```
[the remark this answers is not in the document]
Comment (pyrooka) [reply src/urllib3/util/retry.py]: Yeah, that was copy-paste error, fixed!

File: test/test_retry_deprecated.py [hunk 2]
```diff
@@ -429,6 +432,17 @@ def test_cls_set_default_redirect_headers_blacklist(self, expect_retry_deprecati
...
+        old_setting = Retry.BACKOFF_MAX
+        try:
+            Retry.BACKOFF_MAX = 99
+            retry = Retry()
+            assert retry.DEFAULT_BACKOFF_MAX == 99
+            assert retry.BACKOFF_MAX == 99
+        finally:
+            Retry.BACKOFF_MAX = old_setting
+            assert Retry.BACKOFF_MAX == old_setting
+
```
[the remark this answers is not in the document]
Comment (pyrooka) [reply test/test_retry_deprecated.py:445]: How do you mean? Doesn't the `expect_retry_deprecation` fixture enough to test the warning?

**Review by sethmlarson — APPROVED** [review]
Thanks for this!

**Pull request merged**